### PR TITLE
libosinfo: update to 1.4.0 and add additional dependencies

### PR DESCRIPTION
**Pull request by abusse**
#### Description

This pull request update libosinfo to v1.4.0 and at the same time adds two new ports as dependency that allow libosinfo to return up to date information regarding OSs. This is especially necessary when used with virt-manager as it allows virt-manager to generate an acceptable VM configuration.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.3 18D109
Xcode 10.2 10E125

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/osinfo-db/Portfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                    size    83024
+
+depends_build       port:osinfo-db-tools
+
+use_xz             yes
+use_configure      no
+
+extract {}
+
+build {}
```
Comment (mf2k) [inline sysutils/osinfo-db/Portfile]: So this is basically a database file? If so, add the following line below `maintainers`:

`supported_archs noarch
`

File: sysutils/osinfo-db/Portfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+master_sites        https://releases.pagure.org/libosinfo/
+
+checksums           rmd160  7c1394e42ee5372b92294cf2d7fda713a08abb7d \
+                    sha256  0e31c3f19f2f897dd2ba08d8656fe0bc4abc24f9c1558aff46bb9e9858cbbdb6 \
+                    size    83024
+
+depends_build       port:osinfo-db-tools
+
+use_xz             yes
+use_configure      no
```
Comment (mf2k) [inline sysutils/osinfo-db/Portfile]: Please line up the spacing on the right side arguments with the rest of the portfile.

File: sysutils/osinfo-db/Portfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+extract {}
+
+build {}
+
+destroot {
+	system "osinfo-db-import --root \"${destroot}${prefix}\" --dir \"/usr/share/osinfo\" \"${distpath}/${distfiles}\""
+}
```
Comment (mf2k) [inline sysutils/osinfo-db/Portfile]: Only 4 spaces are needed before the `system` command

File: sysutils/osinfo-db-tools/Portfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                    sha256  7ee0125c6b99b5b9e10d9bf5951ecc8a170278499a778e6be5c61ef10296f723 \
+                    size    519089
+
+depends_lib         path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    port:gettext \
+                    port:json-glib \
+                    port:libarchive \
+                    port:libxslt
+
+depends_build       port:pkgconfig
```
Comment (mf2k) [inline sysutils/osinfo-db-tools/Portfile]: Please move the `depends_build` line before the `depends_lib` line.

File: sysutils/osinfo-db/Portfile (hunk 1 above)
Comment (abusse) [reply sysutils/osinfo-db/Portfile]: I guess it is. I've made the change.

File: sysutils/osinfo-db/Portfile (hunk 2 above)
Comment (abusse) [reply sysutils/osinfo-db/Portfile]: Done.

File: sysutils/osinfo-db/Portfile (hunk 3 above)
Comment (abusse) [reply sysutils/osinfo-db/Portfile]: Done.

File: sysutils/osinfo-db-tools/Portfile (hunk 4 above)
Comment (abusse) [reply sysutils/osinfo-db-tools/Portfile]: Done.

**Comment by mf2k** [comment]
The new commits with the updates should be squashed into the initial commits so that you only have one commit per portfile.

**Comment by abusse** [comment]
Ok, I squashed them. I wasn't completely sure what the integration flow is in this repository.

**Comment by mf2k** [comment]
Let's give Dan some time to comment.

**Comment by mf2k** [comment]
Thank you! Please consider another PR to take over as maintainer.

**Comment by abusse** [comment]
What exactly do I have to do to become a maintainer? Putting my github account into the Portfile or is there a certain process?

**Comment by mf2k** [comment]
All you need to do is add your GitHub handle and email to the `maintainers` line. As maintainer you submit PRs to update it and watch for Trac tickets and try to get them resolved.

**Pull request merged**